cumulusci/cli/cci.py

```diff
@@ -216,33 +217,39 @@ def main(args=None):
         if debug:
             args.remove("--debug")
 
-        # Load CCI config
-        global RUNTIME
-        try:
-            RUNTIME = CliRuntime(load_keychain=False)
-        except Exception as e:
-            handle_exception(e, is_error_command, tempfile_path, debug)
-            sys.exit(1)
+        with DebugMode.set(debug):
+            # Load CCI config into global RUNTIME
+            _load_runtime(is_error_command, tempfile_path, debug)
 
-        RUNTIME.check_cumulusci_version()
+            should_show_stacktraces = RUNTIME.universal_config.cli__show_stacktraces
 
-        should_show_stacktraces = RUNTIME.universal_config.cli__show_stacktraces
+            init_logger(log_requests=debug)
+            # Hand CLI processing over to click, but handle exceptions
+            try:
+                cli(args[1:], standalone_mode=False)
+            except click.Abort:  # Keyboard interrupt
+                show_debug_info() if debug else click.echo("\nAborted!", err=True)
+                sys.exit(1)
+            except Exception as e:
+                if debug:
+                    show_debug_info()
+                else:
+                    handle_exception(
+                        e, is_error_command, tempfile_path, should_show_stacktraces
+                    )
+                sys.exit(1)
+
+
+def _load_runtime(is_error_command, tempfile_path, debug):
+    global RUNTIME
+    try:
+        RUNTIME = CliRuntime(load_keychain=False)
+    except Exception as e:
+        handle_exception(e, is_error_command, tempfile_path, debug)
```

Comment:
Maybe handle_exception should make use of the debug mode now so that it doesn't have to be passed around.

Reply:
Right, and we have a way to turn on should_show_stacktraces without turning on debug mode, okay.